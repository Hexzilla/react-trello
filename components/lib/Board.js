import React, { Component } from 'react'
import './Board.scss'

export default class Board extends Component {
  render () {
    return <div className='board'>
      {this.props.children}
    </div>
  }
}

Board.propTypes = {
  children: React.PropTypes.node
}
