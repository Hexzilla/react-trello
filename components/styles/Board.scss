.board {
  background-color: #23719F;
  overflow: auto;
  padding: 5px;
  font: 14px/18px "Helvetica Neue", Arial, Helvetica, sans-serif;
  color: #393939;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
}

.list {
  background-color: #E3E3E3;
  border-radius: 3px;
  margin: 5px 5px;
  padding: 10px;
  min-width: 220px;

  header {
    margin-bottom: 10px;
    display: flex;
    flex-direction: row;
    align-items: flex-start;

    .title {
      font-weight: bold;
      font-size: 15px;
      line-height: 18px;
      cursor: grab;
      width: 80%;
    }

    .rightContent {
      width: 30%;
      text-align: right;
      font-size: 13px;
    }
  }

  article {
    border-radius: 3px;
    margin: 10px 0px;
  }

  .card {
    background-color: #FFF;
    border-bottom: 1px solid #CCC;
    padding: 6px 8px;
    cursor: pointer;

    header {
      border-bottom: 1px solid #eee;
      padding-bottom: 6px;
    }

    .title {
      font-size: 14px;
    }

    .rightContent {
      font-size: 10px;
    }

    &:hover {
      background-color: #F0F0F0;
    }
  }
}

.detail {
  font-size: 12px;
  color: #939393;
}
