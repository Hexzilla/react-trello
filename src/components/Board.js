import React, {Component, PropTypes} from 'react'
import BoardContainer from './BoardContainer'
import {Provider} from 'react-redux'
import {createStore, applyMiddleware} from 'redux'
import boardReducer from '../reducers/BoardReducer'
import logger from 'redux-logger'

const middlewares = process.env.NODE_ENV === 'development' ? [logger] : []
const store = createStore(boardReducer, applyMiddleware(...middlewares))

export default class Board extends Component {

  render () {
    return <Provider store={store}>
      <BoardContainer {...this.props} />
    </Provider>
  }
}

Board.propTypes = {
  data: PropTypes.object.isRequired,
  onLaneScroll: PropTypes.func,
  onCardClick: PropTypes.func,
  eventBusHandle: PropTypes.func,
  laneSortFunction: PropTypes.func,
  draggable: PropTypes.bool,
  handleDragStart: PropTypes.func,
  handleDragEnd: PropTypes.func,
  onDataChange: PropTypes.func,
  style: PropTypes.object
}
