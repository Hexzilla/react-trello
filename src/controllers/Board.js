import React, {Component} from 'react'
import BoardContainer from './BoardContainer'
import {Provider} from 'react-redux'
import classNames from 'classnames'
import {applyMiddleware, createStore} from 'redux'
import boardReducer from 'reducers/BoardReducer'
import logger from 'redux-logger'
import uuidv1 from 'uuid/v1'
import createTranslate from 'helpers/createTranslate'

const middlewares = process.env.REDUX_LOGGING ? [logger] : []

export default class Board extends Component {
  constructor({id}) {
    super()
    this.store = this.getStore()
    this.id = id || uuidv1()
  }

  getStore = () => {
    //When you create multiple boards, unique stores are created for isolation
    return createStore(boardReducer, applyMiddleware(...middlewares))
  }

  render() {
    const {id, className, components} = this.props
    const allClassNames = classNames('react-trello-board', className || '')
    return (
      <Provider store={this.store}>
        <>
          <components.GlobalStyle />
          <BoardContainer
            id={this.id}
            {...this.props}
            className={allClassNames}
          />
       </>
      </Provider>
    )
  }
}
