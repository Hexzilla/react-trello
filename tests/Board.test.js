import React from 'react'
import renderer from 'react-test-renderer'
import Board from '../components/lib/Board'

describe('<Board/>', () => {
  it('snapshot test', () => {
    const component = renderer.create(
      <Board key='board1'>
        Many boards here
      </Board>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
