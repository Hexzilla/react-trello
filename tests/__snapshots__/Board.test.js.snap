exports[`<Board/> snapshot test 1`] = `
<div
  className="dMcBAq"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }>
  <section
    className="kjauoa"
    draggable={undefined}>
    <header
      className="eQnITk">
      <span
        className="jgmSre">
        Lane1
      </span>
      <span
        className="ggHFfo" />
    </header>
    <div>
      <div
        className="gPtRcD" />
    </div>
  </section>
</div>
`;
